lib/saml2.coffee

```diff
@@ -415,6 +415,17 @@ add_namespaces_to_child_assertions = (xml_string) ->
 parse_authn_response = (saml_response, sp_private_keys, idp_certificates, allow_unencrypted, ignore_signature, require_session_index, cb) ->
   user = {}
 
+  # strip of possible enveloping xml tags:
+  saml_response = saml_response.getElementsByTagNameNS(XMLNS.SAMLP, 'Response')[0] or saml_response
+
+  # check if we have a special case where the complete response is signed (an envelopped signature):
+  if !ignore_signature
+    for cert in idp_certificates or []
+      signed_data_from_complete_saml_response = check_saml_signature(saml_response.toString(), cert)
+      if signed_data_from_complete_saml_response.length is 1 and signed_data_from_complete_saml_response[0] == saml_response.toString()
```

Comment:
When there are multiple Signature nodes in the response, this throws

```js
TypeError: Cannot read property 'length' of null
    at parse_authn_response (/Users/mdolan/Repos/saml-test/node_modules/saml2-js/lib-js/saml2.js:603:50)
```

(because `check_saml_signature` returns `null` https://github.com/Clever/saml2/blob/master/lib/saml2.coffee#L178)

Maybe just need this to be `signed_data_from_complete_saml_response?.length is 1`?

Reply:
Also, is there a reason for the string comparison? There are slight differences in ordering of the fields for me